fetch.bs

```diff
@@ -1876,18 +1876,45 @@ with given <var>strategy</var>, <var>pull</var> action and <var>cancel</var> act
 are optional, run these steps:
 
 <ol>
- <li><p>Let <var>init</var> be a new object.
+ <li><p>Let <var>startAlg</var> be an algorithm that returns <code>undefined</code>.
 
- <li><p>Set <var>init</var>["pull"] to a function that runs <var>pull</var> if <var>pull</var> is
- given.
+ <li><p>Let <var>pullAlg</var> be an algorithm that returns:
 
- <li><p>Set <var>init</var>["cancel"] to a function that runs <var>cancel</var> if
- <var>cancel</var> is given.
+ <dl class=switch>
+ <dt>If <var>pull</var> is given
+ <dd>the result of [=promise-calling=] <var>pull</var>().
+ <dt>Otherwise
+ <dd>[=a promise resolved with=] <code>undefined</code>.
+ </dl>
+
+ <li><p>Let <var>cancelAlg</var> be an algorithm that takes a <var>reason</var> and returns:
+ <dl class=switch>
+ <dt>If <var>cancel</var> is given
+ <dd>the result of [=promise-calling=] <var>cancel</var>(<var>reason</var>).
+ <dt>Otherwise
+ <dd>[=a promise resolved with=] <code>undefined</code>.
+ </dl>
 
- <li><p>Let <var>stream</var> be the result of calling the initial value of {{ReadableStream}} as
- constructor with <var>init</var> and <var>strategy</var> if given.
+ <li><p>Let <var>highWaterMark</var> be:
+ <dl class=switch>
```

Comment:
The need to unpack the strategy argument is unfortunate. If the [reference in HTTP-Network-Fetch](https://fetch.spec.whatwg.org/#ref-for-concept-construct-readablestream%E2%91%A0) is the only usage, I would prefer to change that from the (scary) phrasing:

> Let _strategy_ be an object. The user agent may choose any object.

to something like

> Let _highWaterMark_ be a be a non-negative, non-NaN number, chosen by the user agent.
> Let _sizeAlgorithm_ be an algorithm that accepts chunk objects and returns a number, chosen by the user agent.

Reply:
Yep. I didn't do that because I'm nervous about other callers outside Fetch, but I could leave a note here about the previous state and expect them to update.

Reply:
I've done this and added the note.